apps/recnet-api/src/modules/user/user.controller.ts

```diff
@@ -1,16 +1,195 @@
-import { Controller, Get, Param } from "@nestjs/common";
-import { User as UserModel } from "@prisma/client";
-import UserRepository from "src/database/repository/user.repository";
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  HttpCode,
+  HttpStatus,
+  Patch,
+  Post,
+  Query,
+  UseFilters,
+  UsePipes,
+} from "@nestjs/common";
+import {
+  ApiBearerAuth,
+  ApiCreatedResponse,
+  ApiOkResponse,
+  ApiOperation,
+  ApiTags,
+} from "@nestjs/swagger";
 
-@Controller("user")
+import { Auth, AuthFirebase } from "@recnet-api/utils/auth/auth.decorator";
+import { AuthFirebaseUser, AuthUser } from "@recnet-api/utils/auth/auth.type";
+import { FirebaseUser, User } from "@recnet-api/utils/auth/auth.user.decorator";
+import { RecnetExceptionFilter } from "@recnet-api/utils/filters/recnet.exception.filter";
+import { ZodValidationPipe } from "@recnet-api/utils/pipes/zod.validation.pipe";
+
+import {
+  deleteUserFollowParamsSchema,
+  getUsersParamsSchema,
+  patchUserMeRequestSchema,
+  postUserFollowRequestSchema,
+  postUserMeRequestSchema,
+  postUserValidateHandleRequestSchema,
+  postUserValidateInviteCodeRequestSchema,
+} from "@recnet/recnet-api-model";
+
+import { CreateUserDto } from "./dto/create.user.dto";
+import { FollowUserDto, UnfollowUserDto } from "./dto/follow.user.dto";
+import { QueryUsersDto } from "./dto/query.users.dto";
+import { UpdateUserDto } from "./dto/update.user.dto";
+import {
+  ValidateUserHandleDto,
+  ValidateUserInviteCodeDto,
+} from "./dto/validate.user.dto";
+import { GetUserMeResponse, GetUsersResponse } from "./user.response";
+import { UserService } from "./user.service";
+
+@ApiTags("users")
+@Controller("users")
+@UseFilters(RecnetExceptionFilter)
 export class UserController {
-  constructor(private readonly userRepository: UserRepository) {}
-
-  /* For development */
-  @Get(":handle")
-  public async getUserByHandle(
-    @Param("handle") handle: string
-  ): Promise<Promise<UserModel>> {
-    return this.userRepository.findByHandle(handle);
+  constructor(private readonly userService: UserService) {}
+
+  @ApiOperation({
+    summary: "Get users",
+    description:
+      "Get users with pagination and filter. If no filter is provided, all users are returned.",
+  })
+  @ApiOkResponse({ type: GetUsersResponse })
+  @Get()
+  @UsePipes(new ZodValidationPipe(getUsersParamsSchema))
+  public async getUsers(
+    @Query() dto: QueryUsersDto
+  ): Promise<GetUsersResponse> {
+    const { page, pageSize, ...filter } = dto;
+    return this.userService.getUsers(page, pageSize, filter);
+  }
+
+  @ApiOperation({
+    summary: "User login",
+    description: "Get login user by provider and providerId.",
+  })
+  @ApiBearerAuth()
+  @Get("login")
+  @AuthFirebase()
+  public async login(@FirebaseUser() firebaseUser: AuthFirebaseUser) {
+    const { provider, providerId } = firebaseUser;
+    return this.userService.login(provider, providerId);
+  }
```

Comment:
```suggestion
  public async login(
    @FirebaseUser() firebaseUser: AuthFirebaseUser
  ): Promise<GetUserMeResponse> {
    const { provider, providerId } = firebaseUser;
    const user = await this.userService.login(provider, providerId);
    return { user };
  }
```

The return type is wrong here.